From f1c4937b891005a61341317191905a93bbfaef14 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 2 Feb 2023 13:11:22 +0000
Subject: [PATCH] carotene: don't use -fsched2-use-superblocks

* just as temporary work around for carotene triggering ICE with gcc-13:
opencv/4.6.0-r0/git/3rdparty/carotene/src/flip.cpp:98:1: internal compiler error: in cselib_subst_to_values, at cselib.cc:2148
   98 | }
      | ^
0x1afbcbb internal_error(char const*, ...)
        ???:0
0x6e40e4 fancy_abort(char const*, int, char const*)
        ???:0
0xa12da1 cselib_subst_to_values_from_insn(rtx_def*, machine_mode, rtx_insn*)
        ???:0
0x1a4f9b1 deps_analyze_insn(deps_desc*, rtx_insn*)
        ???:0
0x1a501da sched_analyze(deps_desc*, rtx_insn*, rtx_insn*)
        ???:0
0x1a532f6 schedule_ebb(rtx_insn*, rtx_insn*, bool)
        ???:0
0x1a537e3 schedule_ebbs()
        ???:0
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.

Upstream-Status: Pending
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 3rdparty/carotene/hal/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/3rdparty/carotene/hal/CMakeLists.txt b/3rdparty/carotene/hal/CMakeLists.txt
index 592771c676..0808cbf1c1 100644
--- a/3rdparty/carotene/hal/CMakeLists.txt
+++ b/3rdparty/carotene/hal/CMakeLists.txt
@@ -28,7 +28,7 @@ if(CV_GCC OR CV_CLANG)
   elseif(CV_CLANG)
     list(APPEND TEGRA_COMPILER_FLAGS -fwrapv)
   else()
-    list(APPEND TEGRA_COMPILER_FLAGS -fweb -fwrapv -frename-registers -fsched2-use-superblocks -fsched2-use-traces
+    list(APPEND TEGRA_COMPILER_FLAGS -fweb -fwrapv -frename-registers -fsched2-use-traces
                                      -fsched-stalled-insns-dep=100 -fsched-stalled-insns=2)
   endif()
   if((ANDROID_COMPILER_IS_CLANG OR NOT ANDROID_COMPILER_VERSION VERSION_LESS "4.7") AND ANDROID_NDK_RELEASE STRGREATER "r8d" )
